src/Bicep.LangServer/Server.cs

```diff
@@ -105,13 +106,15 @@ private static void RegisterServices(CreationOptions creationOptions, IServiceCo
             services.AddSingleton<IResourceTypeProvider>(services => creationOptions.ResourceTypeProvider ?? AzResourceTypeProvider.CreateWithAzTypes());
             // We'll use default bicepconfig.json settings during SnippetsProvider creation to avoid errors during language service initialization.
             // We don't do any validation in SnippetsProvider. So using default settings shouldn't be a problem.
-            services.AddSingleton<ISnippetsProvider>(services => creationOptions.SnippetsProvider ?? new SnippetsProvider(fileResolver, new ConfigHelper(null, new FileResolver(), useDefaultConfig: true)));
+            services.AddSingleton<ISnippetsProvider>(services => creationOptions.SnippetsProvider ?? new SnippetsProvider(fileResolver, services.GetRequiredService<IConfigurationManager>().GetBuiltInConfiguration()));
             services.AddSingleton<IFileResolver>(services => fileResolver);
             services.AddSingleton<IFeatureProvider>(services => creationOptions.Features ?? new FeatureProvider());
             services.AddSingleton<IModuleRegistryProvider, DefaultModuleRegistryProvider>();
             services.AddSingleton<IContainerRegistryClientFactory, ContainerRegistryClientFactory>();
             services.AddSingleton<ITemplateSpecRepositoryFactory, TemplateSpecRepositoryFactory>();
             services.AddSingleton<IModuleDispatcher, ModuleDispatcher>();
+            services.AddSingleton<IFileSystem, FileSystem>();
```

Comment:
Nice! Do you think it's worth keeping the `IFileResolver` abstraction around if we have this one? If not, mind creating a tech debt issue to remove it?

Reply:
Yeah I think we can definitely do some refactoring and simplify the abstraction a lot with `IFileSystem`, though we might want to keep some method that `IFileResolver` provides. Created https://github.com/Azure/bicep/issues/4616 to track this.

Reply:
IFileSystem operates on strings instead of URIs btw.

Reply:
Yeah that's something to think about. Maybe we can add some `IFileSystem` extension methods that take URIs and then do `uri.LocalPath`.